nerfstudio/viewer/render_panel.py

```diff
@@ -726,6 +729,144 @@ def _(_: viser.GuiEvent) -> None:
     def _(_) -> None:
         camera_path.show_spline = show_spline_checkbox.value
         camera_path.update_spline()
+    
+    auto_camera_folder = server.gui.add_folder("Automatic Camera Path")
+    with auto_camera_folder:
+        click_position = np.array([1.0, 0.0, 0.0])
+        select_center_button = server.gui.add_button(
+            "Select Center",
+            icon=viser.Icon.CROSSHAIR,
+            hint="Choose center point to generate camera path around.",
+        )
+        
+        @select_center_button.on_click
+        def _(event: viser.GuiEvent) -> None:
+            select_center_button.disabled = True
+
+            @event.client.scene.on_pointer_event(event_type="click")
+            def _(event: viser.ScenePointerEvent) -> None:
+                # Code mostly borrowed from garfield.studio!
+                import torch
+                from nerfstudio.cameras.rays import RayBundle
+                from nerfstudio.field_components.field_heads import FieldHeadNames
+                from nerfstudio.model_components.losses import scale_gradients_by_distance_squared
+
+                origin = torch.tensor(event.ray_origin).view(1, 3)
+                direction = torch.tensor(event.ray_direction).view(1, 3)
+
+                # Get intersection
+                bundle = RayBundle(
+                    origins=origin,
+                    directions=direction,
+                    pixel_area=torch.tensor(0.001).view(1, 1),
+                    camera_indices=torch.tensor(0).view(1, 1),
+                    nears=torch.tensor(0.05).view(1, 1),
+                    fars=torch.tensor(100).view(1, 1),
+                ).to("cuda")
+
+                # Get the distance/depth to the intersection --> calculate 3D position of the click
```

Comment:
This is a good start! but this part should use the more general get_outputs_for_camera() function (a lot of the sampling, scaling stuff is method-specific to nerfacto). As long as the method outputs a 'depth' value it should work with this ray deprojection approach

Reply:
This way, you would render a depth image from the viewer camera and deproject the click point with the intrinsics matrix + rendered depth, meaning it doesn't matter what the rendering backend is.

Reply:
One thing to note is that the 'depth' in splatfacto is actually z-depth and not ray-depth, so the math would need to be different for the two methods

Reply:
Ok that makes sense! Thanks for the feedback appreciate it!!